pyhap/accessory.py

```diff
@@ -193,15 +196,29 @@ def _set_services(self):
             always call the base implementation first, as it reserves IID of
             1 for the Accessory Information service (HAP requirement).
         """
+        info_data = {
+            'Name': self.display_name,
+            'Manufacturer': 'Default Manufacturer',
+            'Model': 'Default Model',
+            'SerialNumber': 'Default SerialNumber'
+        }
+        info_data.update(**self.get_accessory_information())
+
         info_service = get_serv_loader().get("AccessoryInformation")
-        info_service.get_characteristic("Name")\
-                    .set_value(self.display_name, False)
-        info_service.get_characteristic("Manufacturer")\
-                    .set_value("Default-Manufacturer", False)
-        info_service.get_characteristic("Model")\
-                    .set_value("Default-Model", False)
-        info_service.get_characteristic("SerialNumber")\
-                    .set_value("Default-SerialNumber", False)
+        for name, value in info_data.items():
+            try:
+                info_service.get_characteristic(name).set_value(value, False)
+            except KeyError:
+                pass
```

Comment:
Do we really want this behaviour? Imagine I misspell `SerialNUmber` in `get_accessory_information` - I will get no indication that I am referring to a characteristic that does not exist.

I would remove the `try` and fail in case of a missing characteristic. (I am assuming that `get_acc_info` is specifically for configuring the AccessoryInformation service)

Reply:
Yeah, I wasn't sure about this either. Perhaps just having no exception handling will be easiest.